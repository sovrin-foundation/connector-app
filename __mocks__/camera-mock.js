// @flow
import React from 'react'
import { View } from 'react-native'

const constants = {
  Aspect: {},
  BarCodeType: {},
  Type: {},
  CaptureMode: {},
  CaptureTarget: {},
  CaptureQuality: {},
  Orientation: {},
  FlashMode: {},
  TorchMode: {},
}

// We don't need to type check external library
// and we don't need types for mock test stub
class MockCamera extends React.Component<any, void> {
  static constants = constants
  static checkVideoAuthorizationStatus = jest.fn(() => Promise.resolve(true))

  render() {
    return <View>{this.props.children}</View>
  }
}

MockCamera.constants = constants

export default MockCamera
