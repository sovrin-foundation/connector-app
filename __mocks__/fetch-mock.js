import { Headers, Response, Request } from 'node-fetch'

const fetch = jest.fn(() => {})
fetch.Headers = Headers
fetch.Request = Request
fetch.Response = Response

fetch.mockResponse = (body, init) =>
  fetch.mockImplementation(
    () =>
      new Promise((resolve, reject) =>
        process.nextTick(() => resolve(new Response(body, init)))
      )
  )

fetch.mockResponseOnce = (body, init) =>
  fetch.mockImplementationOnce(
    () =>
      new Promise((resolve, reject) =>
        process.nextTick(() => resolve(new Response(body, init)))
      )
  )

fetch.mockResponses = (...responses) =>
  responses.map(([body, init]) =>
    fetch.mockImplementationOnce(
      () =>
        new Promise((resolve, reject) =>
          process.nextTick(() => resolve(new Response(body, init)))
        )
    )
  )

export default fetch
