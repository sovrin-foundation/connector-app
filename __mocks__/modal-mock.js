// @flow
import React from 'react'

const MockModal = (props: any) =>
  React.createElement('Modal', props, props.children)

export default MockModal
