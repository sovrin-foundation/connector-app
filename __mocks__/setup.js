import React from 'react'
import { NativeModules } from 'react-native'
import fetch from './fetch-mock'
import mockCamera from './camera-mock'
import mockModal from './modal-mock'
import mockView from './view-mock'

// mock Math.random for <Loader />
const mockMath = Object.create(global.Math)
mockMath.random = () => 0.1
global.Math = mockMath

jest.mock('NativeAnimatedHelper')

// mock this module to allow react-navigation to mock Linking
jest.mock('Linking', () => ({
  addEventListener: jest.fn(),
  removeEventListener: jest.fn(),
  openURL: jest.fn(() => Promise.resolve()),
  canOpenURL: jest.fn(() => Promise.resolve()),
  getInitialURL: jest.fn(() => Promise.resolve()),
}))

jest.mock('react-navigation', () => ({
  createStackNavigator: jest.fn(() => mockView),
  createTabNavigator: jest.fn(() => mockView),
  TabBarTop: jest.fn(),
  isFocused: true,
  withNavigationFocus: jest.fn(() => mockView),
  NavigationActions: {
    reset() {
      return arguments
    },
    navigate() {
      return arguments
    },
  },
}))

jest.mock('react-native-firebase', () => ({
  notifications: jest.fn(() => ({
    onNotificationDisplayed: jest.fn(() => Promise.resolve()),
    onNotification: jest.fn(() => Promise.resolve()),
    onNotificationOpened: jest.fn(() => Promise.resolve()),
    getInitialNotification: jest.fn(() => Promise.resolve()),
  })),
  messaging: jest.fn(() => ({
    requestPermission: jest.fn(() => Promise.resolve()),
    onTokenRefresh: jest.fn(() => Promise.resolve()),
    getToken: jest.fn(() => Promise.resolve()),
  })),
}))

jest.mock('PixelRatio', () => ({
  get: jest.fn(() => 3), //PixelRatio.get() === 3 then iPhone 6 plus
  roundToNearestPixel: jest.fn(num => num),
}))

jest.mock('react-native-sensitive-info', () => {
  // todo: need to handle empty key-chain case
  const secureStorage = {
    ConnectMeKeyChain: {
      identifier: '26u37qFNUk3QwCUQ9ctest',
      phone: '6428900484',
      seed: 'S1StqtsAL4zkFZxmW57sL75zmufmVFQ3',
      pushComMethod: 'dehVox1KRqM:APA91bFWrJea1avml_ELw2MaH60abydtest',
    },
  }
  return {
    setItem: jest.fn((key, value, st) =>
      Promise.resolve((secureStorage['ConnectMeKeyChain'][key] = value))
    ),
    getItem: jest.fn(key =>
      Promise.resolve(secureStorage['ConnectMeKeyChain'][key])
    ),
    deleteItem: jest.fn(key =>
      Promise.resolve(delete secureStorage['ConnectMeKeyChain'][key])
    ),
  }
})

jest.mock('react-native-animatable', () => ({
  View: 'Animatable.View',
}))

jest.mock('react-native-camera', () => mockCamera)

jest.mock('Dimensions', () => ({
  get: jest.fn(() => ({
    width: 320,
    height: 540,
  })),
}))

jest.mock('react-native-touch-id', () => {
  return {
    authenticate: jest.fn(message => Promise.resolve()),
    isSupported: jest.fn(_ => Promise.resolve()),
  }
})

// jest.mock('react-native-sentry', () => ({
//   Sentry: {
//     install() {},
//     config() {
//       return { install() {} }
//     },
//     captureException(error) {},
//   },
//   SentryLog: {
//     Debug: 1,
//   },
// }))

jest.mock('AlertIOS', () => ({
  alert() {},
}))

global.fetch = fetch

jest.mock('Modal', () => mockModal)

jest.mock('react-native-modal', () => mockModal)

jest.mock('react-native-fetch-blob', () => ({
  fetch: jest.fn((type, url) => Promise.resolve()),
  fs: {
    dirs: {
      DocumentDir: '/var/application/DocumentDir',
    },
    exists: jest.fn(path => Promise.resolve(true)),
    unlink: jest.fn(path => Promise.resolve(true)),
    cp: jest.fn((source, destination) => Promise.resolve(destination)),
  },
}))

jest.mock('WebView', () => 'WebView')

jest.mock('InteractionManager', () => ({
  runAfterInteractions(fn) {
    fn()
  },
  createInteractionHandle: jest.fn(),
  clearInteractionHandle: jest.fn(),
}))

jest.mock('TextInput', () => 'TextInput')

jest.mock('moment', () =>
  jest.fn(date => ({
    format(format) {
      if (format === 'MM/DD/YYYY') {
        return '01/10/2018'
      }

      if (format === 'MMMM YYYY') {
        return 'January 2018'
      }

      return '2018-01-10T04:32:43+05:30'
    },
  }))
)

jest.mock('react-native-share', () => {})

jest.mock('react-native-branch', () => {
  return {
    ADD_TO_CART_EVENT: 'Add to Cart',
    ADD_TO_WISHLIST_EVENT: 'Add to Wishlist',
    PURCHASE_INITIATED_EVENT: 'Purchase Started',
    PURCHASED_EVENT: 'Purchased',
    REGISTER_VIEW_EVENT: 'View',
    SHARE_COMPLETED_EVENT: 'Share Completed',
    SHARE_INITIATED_EVENT: 'Share Started',

    STANDARD_EVENT_ADD_TO_CART: 'ADD_TO_CART',
    STANDARD_EVENT_ADD_TO_WISHLIST: 'ADD_TO_WISHLIST',
    STANDARD_EVENT_VIEW_CART: 'VIEW_CART',
    STANDARD_EVENT_INITIATE_PURCHASE: 'INITIATE_PURCHASE',
    STANDARD_EVENT_ADD_PAYMENT_INFO: 'ADD_PAYMENT_INFO',
    STANDARD_EVENT_PURCHASE: 'PURCHASE',
    STANDARD_EVENT_SPEND_CREDITS: 'SPEND_CREDITS',

    STANDARD_EVENT_SEARCH: 'SEARCH',
    STANDARD_EVENT_VIEW_ITEM: 'VIEW_ITEM',
    STANDARD_EVENT_VIEW_ITEMS: 'VIEW_ITEMS',
    STANDARD_EVENT_RATE: 'RATE',
    STANDARD_EVENT_SHARE: 'SHARE',

    STANDARD_EVENT_COMPLETE_REGISTRATION: 'COMPLETE_REGISTRATION',
    STANDARD_EVENT_COMPLETE_TUTORIAL: 'COMPLETE_TUTORIAL',
    STANDARD_EVENT_ACHIEVE_LEVEL: 'ACHIEVE_LEVEL',
    STANDARD_EVENT_UNLOCK_ACHIEVEMENT: 'UNLOCK_ACHIEVEMENT',

    // don't need any type checks for mocks, because they are not used
    // for type checking in code
    subscribe(cb: any) {},
  }
})

jest.mock('react-native-splash-screen', () => ({
  hide: jest.fn(),
}))

jest.mock('react-native-unique-id', () =>
  jest.fn(() => Promise.resolve('uniqueDeviceId'))
)

jest.mock('react-native-image-crop-picker', () => ({
  openPicker: jest.fn(() =>
    Promise.resolve({ path: '/var/application/DocumentDir/user-avatar.jpeg' })
  ),
}))

jest.mock('react-native-share', () => {})

// Add mock for RNIndy on NativeModules
// we don't want to mock NativeModules the way we are doing for all above
// because then we would have to mock everything else as well of NativeModules
// a lot of other modules depend on it, so we are adding RNIndy
// which is what we need to mock
NativeModules.RNIndy = {
  deserializeConnection: jest.fn(_ => Promise.resolve(1)),
  setWalletItem: jest.fn(_ => Promise.resolve(1)),
  deleteWalletItem: jest.fn(_ => Promise.resolve(1)),
  updateWalletItem: jest.fn(_ => Promise.resolve(1)),
  createWalletKey: jest.fn(_ => Promise.resolve('walletKey')),
  exitAppAndroid: jest.fn(),
}

jest.mock('react-native-version-number', () => ({
  appVersion: '1.0',
  buildVersion: '500',
}))

jest.mock('apptentive-react-native', () => ({
  Apptentive: {
    register: () => Promise.resolve(null),
    presentMessageCenter: jest.fn(),
    onAuthenticationFailed: reason => null,
  },
  ApptentiveConfiguration: () => null,
}))
jest.mock('BackHandler', () => ({
  addEventListener: jest.fn(),
  removeEventListener: jest.fn(),
}))
