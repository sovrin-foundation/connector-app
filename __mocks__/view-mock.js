// @flow
import React from 'react'
import { View } from 'react-native'

class MockView extends React.Component<any, void> {
  render() {
    return <View>{this.props.children}</View>
  }
}

export default MockView
