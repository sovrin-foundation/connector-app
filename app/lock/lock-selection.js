// @flow
import React, { PureComponent } from 'react'
import {
  Image,
  View,
  StyleSheet,
  TouchableHighlight,
  Alert,
  Keyboard,
} from 'react-native'
import { connect } from 'react-redux'
import { select } from 'redux-saga/effects'
import { bindActionCreators } from 'redux'
import type { Store } from '../store/type-store'
import { Container, CustomText, CustomView } from '../components'
import {
  lockPinSetupRoute,
  lockTouchIdSetupRoute,
  switchEnvironmentRoute,
} from '../common/'
import type { LockSelectionProps } from './type-lock'
import {
  OFFSET_1X,
  OFFSET_2X,
  OFFSET_3X,
  OFFSET_5X,
  OFFSET_6X,
  OFFSET_7X,
  color,
  isiPhone5,
  PIN_CODE_BORDER_BOTTOM,
} from '../common/styles/constant'
import { switchErrorAlerts } from '../store/config-store'
import {
  disableDevMode,
  longPressedInLockSelectionScreen,
  pressedOnOrInLockSelectionScreen,
} from './lock-store'
import { safeToDownloadSmsInvitation } from '../sms-pending-invitation/sms-pending-invitation-store'

export class LockSelection extends PureComponent<LockSelectionProps, void> {
  constructor(props: LockSelectionProps) {
    super(props)
    Keyboard.dismiss()
  }

  goTouchIdSetup = () => {
    if (this.props.navigation.isFocused()) {
      this.props.navigation.navigate(lockTouchIdSetupRoute)
      this.props.safeToDownloadSmsInvitation()
    }
  }

  goPinCodeSetup = () => {
    this.props.navigation.navigate(lockPinSetupRoute)
    this.props.safeToDownloadSmsInvitation()
  }

  _onLongPressButton = () => {
    this.props.longPressedInLockSelectionScreen()
  }

  _onTextPressButton = () => {
    this.props.pressedOnOrInLockSelectionScreen()
  }

  componentWillReceiveProps(nextProps: LockSelectionProps) {
    if (nextProps.showDevMode) {
      Alert.alert(
        'Developer Mode',
        'you are enabling developer mode and it will delete all existing data. Are you sure?',
        [
          {
            text: 'Cancel',
            style: 'cancel',
            onPress: () => nextProps.disableDevMode(),
          },
          {
            text: 'OK',
            onPress: () =>
              nextProps.navigation.navigate(switchEnvironmentRoute),
          },
        ]
      )
    }
  }

  render() {
    return (
      <Container tertiary style={[style.pinSelectionContainer]}>
        <CustomView
          onPress={this.props.switchErrorAlerts}
          onLongPress={this._onLongPressButton}
        >
          <CustomText h5 bg="tertiary" tertiary semiBold center>
            Choose how to unlock App
          </CustomText>
        </CustomView>
        <CustomView style={[style.messageText]}>
          <CustomText h5 bg="tertiary" tertiary bold center>
            This application must be protected by Biometrics or a passcode at
            all times.
          </CustomText>
        </CustomView>
        <Container spaceAround>
          <CustomView
            center
            fifth
            shadow
            testID="touch-id-selection"
            style={[style.touchIdPinContainer]}
            onPress={this.goTouchIdSetup}
            onLongPress={this._onLongPressButton}
          >
            <Image
              style={style.fingerPrintIcon}
              source={require('../images/icon_fingerPrint.png')}
            />
            <CustomText
              h5
              semiBold
              center
              tertiary
              transparentBg
              testID="use-touch-id-text"
              onPress={this.goTouchIdSetup}
              onLongPress={this._onLongPressButton}
            >
              Use Biometrics
            </CustomText>
          </CustomView>
          <CustomView
            testID="lock-selection-or-text"
            onLongPress={this._onLongPressButton}
            onPress={this._onTextPressButton}
            debounceAction={false}
          >
            <CustomText h4 bg="tertiary" tertiary transparentBg thick center>
              or
            </CustomText>
          </CustomView>
          <CustomView
            fifth
            shadow
            testID="pin-code-selection"
            style={[style.touchIdPinContainer]}
            onPress={this.goPinCodeSetup}
            onLongPress={this._onLongPressButton}
          >
            <CustomView row center style={[style.pinContainer]}>
              <CustomView style={[style.pin]}>
                <CustomText h4 thick center bg="fifth">
                  1
                </CustomText>
              </CustomView>
              <CustomView style={[style.pin]}>
                <CustomText h4 thick center bg="fifth">
                  2
                </CustomText>
              </CustomView>
              <CustomView style={[style.pin]}>
                <CustomText h4 thick center bg="fifth">
                  3
                </CustomText>
              </CustomView>
              <CustomView style={[style.pin]}>
                <CustomText h4 thick center bg="fifth">
                  4
                </CustomText>
              </CustomView>
              <CustomView style={[style.pin]}>
                <CustomText h4 thick center bg="fifth">
                  5
                </CustomText>
              </CustomView>
              <CustomView style={[style.pin]}>
                <CustomText h4 thick center bg="fifth">
                  6
                </CustomText>
              </CustomView>
            </CustomView>
            <CustomText
              h5
              semiBold
              center
              tertiary
              transparentBg
              style={[style.usePinText]}
              onPress={this.goPinCodeSetup}
              testID="use-pass-code-text"
              onLongPress={this._onLongPressButton}
            >
              Use Passcode
            </CustomText>
          </CustomView>
        </Container>
      </Container>
    )
  }
}

const mapStateToProps = ({ lock }: Store) => {
  return {
    showDevMode: lock.showDevMode,
  }
}

const mapDispatchToProps = dispatch =>
  bindActionCreators(
    {
      switchErrorAlerts,
      longPressedInLockSelectionScreen,
      pressedOnOrInLockSelectionScreen,
      disableDevMode,
      safeToDownloadSmsInvitation,
    },
    dispatch
  )

export default connect(mapStateToProps, mapDispatchToProps)(LockSelection)

const style = StyleSheet.create({
  pinSelectionContainer: {
    paddingTop: OFFSET_3X,
    paddingBottom: isiPhone5 ? OFFSET_2X : OFFSET_6X,
    paddingHorizontal: OFFSET_2X,
  },
  messageText: {
    paddingHorizontal: isiPhone5 ? 0 : OFFSET_5X / 2,
    paddingTop: isiPhone5 ? OFFSET_5X / 2 : OFFSET_5X,
    paddingBottom: isiPhone5 ? OFFSET_3X / 2 : OFFSET_7X / 2,
  },
  touchIdPinContainer: {
    paddingTop: OFFSET_1X / 2,
    paddingHorizontal: OFFSET_2X,
    paddingBottom: OFFSET_2X,
    borderRadius: 13,
    marginHorizontal: OFFSET_3X,
  },
  fingerPrintIcon: {
    marginVertical: OFFSET_3X / 2,
    height: 60,
    width: 60,
  },
  pinContainer: {
    marginVertical: OFFSET_3X,
    marginBottom: OFFSET_5X / 2,
  },
  pin: {
    borderBottomColor: color.bg.fifth.font.primary,
    borderBottomWidth: PIN_CODE_BORDER_BOTTOM,
    marginRight: OFFSET_1X / 2,
    paddingHorizontal: OFFSET_1X / 2,
    paddingBottom: OFFSET_1X,
  },
  usePinText: {
    lineHeight: 22,
    paddingBottom: OFFSET_1X / 2,
  },
})
