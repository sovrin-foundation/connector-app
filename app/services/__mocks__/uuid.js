// @flow
import { defaultUUID } from '../../../__mocks__/static-data'

export function uuid() {
  return defaultUUID
}
