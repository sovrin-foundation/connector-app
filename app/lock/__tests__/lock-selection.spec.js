// @flow
import 'react-native'
import React from 'react'
import renderer from 'react-test-renderer'
import { Provider } from 'react-redux'
import { LockSelection } from '../lock-selection'
import { getStore, getNavigation } from '../../../__mocks__/static-data'

function getLockSelectionProps() {
  return {
    switchErrorAlerts: jest.fn(),
    longPressedInLockSelectionScreen: jest.fn(),
    pressedOnOrInLockSelectionScreen: jest.fn(),
    disableDevMode: jest.fn(),
    safeToDownloadSmsInvitation: jest.fn(),
  }
}

const navigation = getNavigation()
const store = getStore()

let componentInstance: LockSelection
let tree
let wrapper
let props

describe('app lock selection page should', () => {
  beforeEach(() => {
    props = getLockSelectionProps()
    wrapper = renderer.create(
      <Provider store={store}>
        <LockSelection {...props} showDevMode={false} navigation={navigation} />
      </Provider>
    )
    tree = wrapper.toJSON()
    componentInstance = wrapper.root.findByType(LockSelection).instance
  })

  it('render properly', () => {
    expect(tree).toMatchSnapshot()
  })

  it('should be able to call longPressedInLockSelectionScreen action', () => {
    componentInstance._onLongPressButton()
    expect(props.longPressedInLockSelectionScreen).toHaveBeenCalled()
  })

  it('should be able to call pressedOnOrInLockSelectionScreen press action', () => {
    componentInstance._onTextPressButton()
    expect(props.pressedOnOrInLockSelectionScreen).toHaveBeenCalled()
  })

  it('call safeToDownloadSmsInvitation if setup touchId', () => {
    componentInstance.goTouchIdSetup()
    expect(props.safeToDownloadSmsInvitation).toHaveBeenCalled()
  })

  it('call safeToDownloadSmsInvitation if setup pass code', () => {
    componentInstance.goPinCodeSetup()
    expect(props.safeToDownloadSmsInvitation).toHaveBeenCalled()
  })
})
