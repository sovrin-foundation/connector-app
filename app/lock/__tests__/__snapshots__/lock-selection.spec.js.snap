// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app lock selection page should render properly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#ffffff",
      "flex": 1,
      "paddingBottom": 20,
      "paddingHorizontal": 20,
      "paddingTop": 30,
    }
  }
  tertiary={true}
>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={
      Object {
        "bottom": 20,
        "left": 20,
        "right": 20,
        "top": 20,
      }
    }
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={Object {}}
    testID={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "fontSize": 17,
          },
          Object {
            "fontFamily": "Lato",
          },
          Object {
            "backgroundColor": "#ffffff",
            "color": "#777777",
          },
          Object {
            "fontWeight": "600",
          },
          Object {
            "textAlign": "center",
          },
          null,
          null,
          null,
          null,
          null,
          null,
          null,
          null,
          null,
        ]
      }
    >
      Choose how to unlock App
    </Text>
  </View>
  <View
    style={
      Object {
        "paddingBottom": 15,
        "paddingHorizontal": 0,
        "paddingTop": 25,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "fontSize": 17,
          },
          Object {
            "fontFamily": "Lato",
          },
          Object {
            "backgroundColor": "#ffffff",
            "color": "#777777",
          },
          Object {
            "fontWeight": "bold",
          },
          Object {
            "textAlign": "center",
          },
          null,
          null,
          null,
          null,
          null,
          null,
          null,
          null,
          null,
        ]
      }
    >
      This application must be protected by Biometrics or a passcode at all times.
    </Text>
  </View>
  <View
    spaceAround={true}
    style={
      Object {
        "flex": 1,
        "justifyContent": "space-around",
      }
    }
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel="touch-id-selection-touchable"
      accessibilityTraits={undefined}
      accessible={false}
      center={true}
      fifth={true}
      hitSlop={
        Object {
          "bottom": 20,
          "left": 20,
          "right": 20,
          "top": 20,
        }
      }
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      shadow={true}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#ffffff",
          "borderRadius": 13,
          "elevation": 10,
          "justifyContent": "center",
          "marginHorizontal": 30,
          "paddingBottom": 20,
          "paddingHorizontal": 20,
          "paddingTop": 5,
          "shadowColor": "rgba(0, 0, 0, 0.2)",
          "shadowOffset": Object {
            "height": 10,
            "width": 0,
          },
          "shadowOpacity": 1,
          "shadowRadius": 30,
        }
      }
      testID="touch-id-selection-touchable"
    >
      <Image
        source={1}
        style={
          Object {
            "height": 60,
            "marginVertical": 15,
            "width": 60,
          }
        }
      />
      <Text
        accessibilityLabel="use-touch-id-text"
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        onLongPress={[Function]}
        onPress={[Function]}
        style={
          Array [
            Object {
              "fontSize": 17,
            },
            Object {
              "fontFamily": "Lato",
            },
            Object {
              "backgroundColor": "#333333",
              "color": "#333333",
            },
            Object {
              "fontWeight": "600",
            },
            Object {
              "textAlign": "center",
            },
            Object {
              "backgroundColor": "transparent",
            },
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            null,
          ]
        }
        testID="use-touch-id-text"
      >
        Use Biometrics
      </Text>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel="lock-selection-or-text-touchable"
      accessibilityTraits={undefined}
      accessible={false}
      debounceAction={false}
      hitSlop={
        Object {
          "bottom": 20,
          "left": 20,
          "right": 20,
          "top": 20,
        }
      }
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={Object {}}
      testID="lock-selection-or-text-touchable"
    >
      <Text
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "fontSize": 20,
            },
            Object {
              "fontFamily": "Lato",
            },
            Object {
              "backgroundColor": "#ffffff",
              "color": "#777777",
            },
            Object {
              "fontWeight": "800",
            },
            Object {
              "textAlign": "center",
            },
            Object {
              "backgroundColor": "transparent",
            },
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            null,
          ]
        }
      >
        or
      </Text>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel="pin-code-selection-touchable"
      accessibilityTraits={undefined}
      accessible={false}
      fifth={true}
      hitSlop={
        Object {
          "bottom": 20,
          "left": 20,
          "right": 20,
          "top": 20,
        }
      }
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      shadow={true}
      style={
        Object {
          "backgroundColor": "#ffffff",
          "borderRadius": 13,
          "elevation": 10,
          "marginHorizontal": 30,
          "paddingBottom": 20,
          "paddingHorizontal": 20,
          "paddingTop": 5,
          "shadowColor": "rgba(0, 0, 0, 0.2)",
          "shadowOffset": Object {
            "height": 10,
            "width": 0,
          },
          "shadowOpacity": 1,
          "shadowRadius": 30,
        }
      }
      testID="pin-code-selection-touchable"
    >
      <View
        center={true}
        row={true}
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "center",
            "marginBottom": 25,
            "marginVertical": 30,
          }
        }
      >
        <View
          style={
            Object {
              "borderBottomColor": "#777777",
              "borderBottomWidth": 4,
              "marginRight": 5,
              "paddingBottom": 10,
              "paddingHorizontal": 5,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={false}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "fontSize": 20,
                },
                Object {
                  "fontFamily": "Lato",
                },
                Object {
                  "backgroundColor": "#ffffff",
                  "color": "#777777",
                },
                Object {
                  "fontWeight": "800",
                },
                Object {
                  "textAlign": "center",
                },
                null,
                null,
                null,
                null,
                null,
                null,
                null,
                null,
                null,
              ]
            }
          >
            1
          </Text>
        </View>
        <View
          style={
            Object {
              "borderBottomColor": "#777777",
              "borderBottomWidth": 4,
              "marginRight": 5,
              "paddingBottom": 10,
              "paddingHorizontal": 5,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={false}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "fontSize": 20,
                },
                Object {
                  "fontFamily": "Lato",
                },
                Object {
                  "backgroundColor": "#ffffff",
                  "color": "#777777",
                },
                Object {
                  "fontWeight": "800",
                },
                Object {
                  "textAlign": "center",
                },
                null,
                null,
                null,
                null,
                null,
                null,
                null,
                null,
                null,
              ]
            }
          >
            2
          </Text>
        </View>
        <View
          style={
            Object {
              "borderBottomColor": "#777777",
              "borderBottomWidth": 4,
              "marginRight": 5,
              "paddingBottom": 10,
              "paddingHorizontal": 5,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={false}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "fontSize": 20,
                },
                Object {
                  "fontFamily": "Lato",
                },
                Object {
                  "backgroundColor": "#ffffff",
                  "color": "#777777",
                },
                Object {
                  "fontWeight": "800",
                },
                Object {
                  "textAlign": "center",
                },
                null,
                null,
                null,
                null,
                null,
                null,
                null,
                null,
                null,
              ]
            }
          >
            3
          </Text>
        </View>
        <View
          style={
            Object {
              "borderBottomColor": "#777777",
              "borderBottomWidth": 4,
              "marginRight": 5,
              "paddingBottom": 10,
              "paddingHorizontal": 5,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={false}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "fontSize": 20,
                },
                Object {
                  "fontFamily": "Lato",
                },
                Object {
                  "backgroundColor": "#ffffff",
                  "color": "#777777",
                },
                Object {
                  "fontWeight": "800",
                },
                Object {
                  "textAlign": "center",
                },
                null,
                null,
                null,
                null,
                null,
                null,
                null,
                null,
                null,
              ]
            }
          >
            4
          </Text>
        </View>
        <View
          style={
            Object {
              "borderBottomColor": "#777777",
              "borderBottomWidth": 4,
              "marginRight": 5,
              "paddingBottom": 10,
              "paddingHorizontal": 5,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={false}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "fontSize": 20,
                },
                Object {
                  "fontFamily": "Lato",
                },
                Object {
                  "backgroundColor": "#ffffff",
                  "color": "#777777",
                },
                Object {
                  "fontWeight": "800",
                },
                Object {
                  "textAlign": "center",
                },
                null,
                null,
                null,
                null,
                null,
                null,
                null,
                null,
                null,
              ]
            }
          >
            5
          </Text>
        </View>
        <View
          style={
            Object {
              "borderBottomColor": "#777777",
              "borderBottomWidth": 4,
              "marginRight": 5,
              "paddingBottom": 10,
              "paddingHorizontal": 5,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={false}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "fontSize": 20,
                },
                Object {
                  "fontFamily": "Lato",
                },
                Object {
                  "backgroundColor": "#ffffff",
                  "color": "#777777",
                },
                Object {
                  "fontWeight": "800",
                },
                Object {
                  "textAlign": "center",
                },
                null,
                null,
                null,
                null,
                null,
                null,
                null,
                null,
                null,
              ]
            }
          >
            6
          </Text>
        </View>
      </View>
      <Text
        accessibilityLabel="use-pass-code-text"
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        onLongPress={[Function]}
        onPress={[Function]}
        style={
          Array [
            Object {
              "fontSize": 17,
            },
            Object {
              "fontFamily": "Lato",
            },
            Object {
              "backgroundColor": "#333333",
              "color": "#333333",
            },
            Object {
              "fontWeight": "600",
            },
            Object {
              "textAlign": "center",
            },
            Object {
              "backgroundColor": "transparent",
            },
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            Object {
              "lineHeight": 22,
              "paddingBottom": 5,
            },
          ]
        }
        testID="use-pass-code-text"
      >
        Use Passcode
      </Text>
    </View>
  </View>
</View>
`;
